src/PSR7/Validators/BodyValidator/MultipartValidation.php

```diff
@@ -0,0 +1,187 @@
+                        throw InvalidHeaders::becauseOfMissingRequiredHeaderMupripart($partName, $headerName, $addr);
+                    }
+
+                    $validator = new SchemaValidator($this->detectValidationStrategy($message));
+                    try {
+                        $validator->validate($headerValue, $headerSchema);
+                    } catch (SchemaMismatch $e) {
+                        throw InvalidHeaders::becauseValueDoesNotMatchSchemaMultipart($partName, $headerName, $headerValue, $addr, $e);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Prepare a Multipart message body (a set of normal parts) for validation against a schema
+     *
+     * @see https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html
+     * @see https://swagger.io/docs/specification/describing-request-body/multipart-requests/
+     *
+     * @return mixed[]
+     *
+     * @throws InvalidBody
+     * @throws TypeMismatch
+     */
+    private function parseMultipartData(OperationAddress $addr, StreamedPart $document) : array
+    {
+        $multipartData = []; // a buffer to fill up with message parts
+        foreach ($document->getParts() as $i => $part) {
+            $partContentType = $part->getHeader('Content-Type');
+
+            if (preg_match('#^application/.*json$#', $partContentType)) {
+                $partBody = json_decode($part->getBody(), true);
+                if (json_last_error() !== JSON_ERROR_NONE) {
+                    throw InvalidBody::becauseBodyIsNotValidJson(json_last_error_msg(), $addr);
+                }
+            } else {
+                $partBody = $part->getBody();
+            }
+
+            // if name is not set, it should be validated with "additionalProperties" keyword
+            $multipartData[$part->getName() ?? '____' . $i] = $partBody;
+        }
+
+        return $multipartData;
+    }
+
+    private function detectEncondingContentType(Encoding $encoding, StreamedPart $part, Schema $partSchema) : ?string
+    {
+        $contentType = $encoding->contentType;
+
+        if (! $contentType) {
+            // fallback strategy:
+            // @see https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#encoding-object
+            // @see https://swagger.io/docs/specification/describing-request-body/multipart-requests/
+            //
+            // Default value depends on the property type: for string with format being binary – application/octet-stream;
+            // for other primitive types – text/plain; for object - application/json; for array – the default is defined based on the inner type.
+            // The value can be a specific media type (e.g. application/json), a wildcard media type (e.g. image/*),
+            // or a comma-separated list of the two types.
+            if ($partSchema->type === 'string') {
+                if (in_array($partSchema->format, ['binary', 'base64'])) {
```

Comment:
are you sure base64 is counted as binary format, not as plaintext? I'm not really sure, thus asking to confirm this

Reply:
Yeah, I took this from here: https://swagger.io/docs/specification/describing-request-body/multipart-requests/ (look up "base64" to find the place)